### PR TITLE
feat: (macos) add vibrancy state parameter

### PR DESCRIPTION
**Pull request by aizcutei**
A solution for #59.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -152,11 +152,12 @@ pub fn clear_mica(window: impl raw_window_handle::HasRawWindowHandle) -> Result<
 pub fn apply_vibrancy(
     window: impl raw_window_handle::HasRawWindowHandle,
     #[allow(unused)] effect: NSVisualEffectMaterial,
+    #[allow(unused)] state: NSVisualEffectState,
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion
    #[allow(unused)] state: Option<NSVisualEffectState>,
```

File: src/macos.rs [hunk 2]
```diff
@@ -107,7 +117,7 @@ mod internal {
 
             blurred_view.setMaterial_(m);
             blurred_view.setBlendingMode_(NSVisualEffectBlendingMode::BehindWindow);
-            blurred_view.setState_(NSVisualEffectState::FollowsWindowActiveState);
+            blurred_view.setState_(state);
```
Comment (amrbashir) [inline src/macos.rs]:
```suggestion
            blurred_view.setState_(state.unwrap_or(NSVisualEffectState::FollowsWindowActiveState));
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks but we need to make this optional and provide a fallback, make sure to document the default behavior in the function documentation.

Also please add a change file in `.changes` folder with `minor` bump

File: src/macos.rs [hunk 3]
```diff
@@ -76,7 +86,7 @@ mod internal {
     use crate::Error;
 
     #[allow(deprecated)]
-    pub fn apply_vibrancy(window: id, appearance: NSVisualEffectMaterial) -> Result<(), Error> {
+    pub fn apply_vibrancy(window: id, appearance: NSVisualEffectMaterial, state: NSVisualEffectState) -> Result<(), Error> {
```
Comment (amrbashir) [inline src/macos.rs]:
```suggestion
    pub fn apply_vibrancy(window: id, appearance: NSVisualEffectMaterial, state: Option<NSVisualEffectState>) -> Result<(), Error> {
```

**Comment by aizcutei** [comment]
I updated some codes and added a change file.
Thanks for your guidance.

File: .changes/state_control.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"window-vibrancy": "minor"
+---
+
+Expose vibrancy state control for macos, default is follow window active state.
```
Comment (amrbashir) [inline .changes/state_control.md:5]: Add a 3rd argument to `apply_vibrancy` to specify the vibrancy state, follows the window active state by default.

Comment (amrbashir) [inline .changes/state_control.md]:
```suggestion
Add a 3rd argument to `apply_vibrancy()` to control the vibrancy state, follows window active state by default.
```

File: README.md [hunk 5]
```diff
@@ -28,7 +28,7 @@ apply_blur(&window, Some((18, 18, 18, 125))).expect("Unsupported platform! 'appl
 | `apply_blur`&`clear_blur`         | Windows  7/10/11      |       |
 | `apply_acrylic`&`clear_acrylic`   | Windows 10/11         | has bad performance when resizing/dragging the window on Windows 10 v1903+ and Windows 11 build 22000. |
 | `apply_mica`&`clear_mica`         | Windows 11            |       |
-| `apply_vibrancy`                  | macOS 10.10 and newer |       |
+| `apply_vibrancy`                  | macOS 10.10 and newer | `state` is used to control vibrancy state, for default please use `None` and it will follow the window active state. |
```
Comment (amrbashir) [inline README.md]: revert this change

File: README.md [hunk 6]
```diff
@@ -12,10 +12,10 @@ Make your windows vibrant.
 ## Example
 
 ```rs
-use window_vibrancy::{apply_blur, apply_vibrancy, NSVisualEffectMaterial};
+use window_vibrancy::{apply_blur, apply_vibrancy, NSVisualEffectMaterial, NSVisualEffectState};
 
 #[cfg(target_os = "macos")]
-apply_vibrancy(&window, NSVisualEffectMaterial::HudWindow).expect("Unsupported platform! 'apply_vibrancy' is only supported on macOS");
+apply_vibrancy(&window, NSVisualEffectMaterial::HudWindow, Some(NSVisualEffectState::Active)).expect("Unsupported platform! 'apply_vibrancy' is only supported on macOS");
```
Comment (amrbashir) [inline README.md]:
```suggestion
apply_vibrancy(&window, NSVisualEffectMaterial::HudWindow, None).expect("Unsupported platform! 'apply_vibrancy' is only supported on macOS");
```

File: README.md [hunk 7]
```diff
@@ -12,10 +12,10 @@ Make your windows vibrant.
 ## Example
 
 ```rs
-use window_vibrancy::{apply_blur, apply_vibrancy, NSVisualEffectMaterial};
+use window_vibrancy::{apply_blur, apply_vibrancy, NSVisualEffectMaterial, NSVisualEffectState};
```
Comment (amrbashir) [inline README.md]:
```suggestion
use window_vibrancy::{apply_blur, apply_vibrancy, NSVisualEffectState};
```

File: src/lib.rs [hunk 8]
```diff
@@ -11,11 +11,11 @@
 //! # Example
 //!
 //! ```no_run
-//! use window_vibrancy::{apply_vibrancy, apply_blur, NSVisualEffectMaterial};
+//! use window_vibrancy::{apply_vibrancy, apply_blur, NSVisualEffectMaterial, NSVisualEffectState};
```
Comment (amrbashir) [inline src/lib.rs]:
```suggestion
//! use window_vibrancy::{apply_vibrancy, apply_blur, NSVisualEffectMaterial};
```

File: src/macos.rs [hunk 9]
```diff
@@ -105,9 +118,14 @@ mod internal {
                 NSVisualEffectView::initWithFrame_(NSVisualEffectView::alloc(nil), bounds);
             blurred_view.autorelease();
 
+            let visual_state = match state {
+                Some(s) => s,
+                None => NSVisualEffectState::FollowsWindowActiveState,
+            };
+
```
Comment (amrbashir) [inline src/macos.rs]: revert this

File: src/macos.rs [hunk 10]
```diff
@@ -105,9 +118,14 @@ mod internal {
...
             blurred_view.setMaterial_(m);
             blurred_view.setBlendingMode_(NSVisualEffectBlendingMode::BehindWindow);
-            blurred_view.setState_(NSVisualEffectState::FollowsWindowActiveState);
+            blurred_view.setState_(visual_state);
```
Comment (amrbashir) [inline src/macos.rs]:
```suggestion
            blurred_view.setState_(state.unwrap_or(NSVisualEffectState::FollowsWindowActiveState));
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
also update `examples/tao.rs` and `examples/winit.rs` and run `cargo fmt --all` before commiting

**Review by amrbashir — APPROVED** [review]
Thank you, I will merge once CI passes

**Pull request merged**